two_factor/views/utils.py

```diff
@@ -1,15 +1,46 @@
 import logging
 
+from django.conf import settings
 from django.core.exceptions import SuspiciousOperation
+from django.core.mail import EmailMessage
 from django.utils.decorators import method_decorator
 from django.utils.translation import ugettext as _
 from formtools.wizard.forms import ManagementForm
 from formtools.wizard.storage.session import SessionStorage
 from formtools.wizard.views import SessionWizardView
 
+from two_factor.templatetags.device_format import agent_format
+
 logger = logging.getLogger(__name__)
 
 
+def send_new_device(request):
+    # prevent circular imports
+    from two_factor import signals
+    signals.login_alert(sender=None, request=request)
+
+
+def send_user_verified(request, user, device):
+    # prevent circular imports
+    from two_factor import signals
+    signals.user_verified.send(sender=None, request=request,
+                               user=user, device=device)
+
+
+def login_alerts(request):
```

Comment:
This really should not be part of this PR. It's nothing to do with skipping 2fa and assumes the user has an email address.